pymc_marketing/mmm/lift_test.py

```diff
@@ -0,0 +1,366 @@
+"""Lift test functions for the MMM."""
+
+from functools import partial
+from typing import Callable, Optional
+
+import numpy as np
+import pandas as pd
+import pymc as pm
+import pytensor.tensor as pt
+from numpy import typing as npt
+
+from pymc_marketing.mmm.transformers import logistic_saturation
+from pymc_marketing.mmm.utils import michaelis_menten
+
+
+class MissingLiftTestError(Exception):
+    def __init__(self, missing_values: npt.NDArray[np.int_]) -> None:
+        self.missing_values = missing_values
+        super().__init__(
+            f"Some lift test values are not in the model: {missing_values}"
+        )
+
+
+def _lift_test_index(
+    lift_values: np.ndarray, model_values: np.ndarray
+) -> npt.NDArray[np.int_]:
+    same_value = lift_values[:, None] == model_values
+    if not (same_value.sum(axis=1) == 1).all():
+        missing_values = np.argwhere(same_value.sum(axis=1) == 0).flatten()
+        raise MissingLiftTestError(missing_values)
+
+    return np.argmax(same_value, axis=1)
+
+
+Index = npt.NDArray[np.int_]
+Indices = dict[str, Index]
+
+
+def lift_test_indices(df_lift_test: pd.DataFrame, model: pm.Model) -> Indices:
+    """Get the indices of the lift test results in the model.
+
+    Assumes any column in the DataFrame is a coordinate in the model with the
+    same name.
+
+    Args:
+        df_lift_test: DataFrame with lift test results.
+        model: PyMC model with date and channel coordinates.
+
+    Returns:
+        Dictionary of indices for the lift test results in the model.
```

Comment:
I think we are following the numpy docstring format right? https://numpydoc.readthedocs.io/en/latest/format.html